src/index/Index.cpp

```diff
@@ -210,43 +234,71 @@ void Index::passTsvFileIntoIdVector(const string& tsvFile, ExtVec& data,
 
 // _____________________________________________________________________________
 size_t Index::passNTriplesFileForVocabulary(const string& ntFile,
-                                            bool onDiskLiterals) {
-  LOG(INFO) << "Making pass over NTriples " << ntFile << " for vocabulary."
-            << std::endl;
-  array<string, 3> spo;
+                                            bool onDiskLiterals, size_t linesPerPartial) {
+  array<string, 3> spo; 
   NTriplesParser p(ntFile);
   ad_utility::HashSet<string> items;
-  size_t i = 0;
+  size_t i = 0; 
+  size_t numFiles = 0; 
   while (p.getLine(spo)) {
     if (ad_utility::isXsdValue(spo[2])) {
       spo[2] = ad_utility::convertValueLiteralToIndexWord(spo[2]);
-    }
+    }    
     if (onDiskLiterals && isLiteral(spo[2]) && shouldBeExternalized(spo[2])) {
       spo[2] = string({EXTERNALIZED_LITERALS_PREFIX}) + spo[2];
+    }    
+    
+    // Duplicated in pass for Id Vector, externalize to function
+    for (size_t k = 0; k < 3; ++k) {
+      items.insert(spo[k]);
     }
-    items.insert(spo[0]);
-    items.insert(spo[1]);
-    items.insert(spo[2]);
+
     ++i;
     if (i % 10000000 == 0) {
+      std::cout << "Lines processed: " << i << '\n';
+    }
+
+    if (i % linesPerPartial == 0) {
       LOG(INFO) << "Lines processed: " << i << '\n';
+      std::cout << "writing partial vocab no. " << numFiles << std::endl;
+      Vocabulary vocab;
+      vocab.createFromSet(items);
+      vocab.writeToBinaryFileForMerging(_onDiskBase + "partialVocabulary" + std::to_string(numFiles));
+      items.clear();
+      numFiles++;
     }
   }
+  // write Remainder
+  //
+  LOG(INFO) << "Lines processed: " << i << '\n';
+  if (items.size() > 0) {
+    // write remainder
+    
+    std::cout << "writing partial vocab no. " << numFiles << std::endl;
+    Vocabulary vocab;
+    vocab.createFromSet(items);
+    vocab.writeToBinaryFileForMerging(_onDiskBase + "partialVocabulary" + std::to_string(numFiles));
+    items.clear();
+    numFiles++;
+  }
+  LOG(INFO) << "Merging vocabulary\n";
+  mergeVocabulary(_onDiskBase, numFiles);
   LOG(INFO) << "Pass done.\n";
-  _vocab.createFromSet(items);
-  items.clear();
   return i;
 }
 
 // _____________________________________________________________________________
 void Index::passNTriplesFileIntoIdVector(const string& ntFile, ExtVec& data,
-                                         bool onDiskLiterals) {
+                                         bool onDiskLiterals, size_t linesPerPartial) {
   LOG(INFO) << "Making pass over NTriples " << ntFile
             << " and creating stxxl vector.\n";
   array<string, 3> spo;
   NTriplesParser p(ntFile);
-  google::sparse_hash_map<string, Id> vocabMap = _vocab.asMap();
+  LOG(INFO) << "Reading partial vocab ...\n";
+  google::sparse_hash_map<string, Id> vocabMap = vocabMapFromPartialIndexedFile(_onDiskBase + "partialVocabulary0");
```

Comment:
I think we should stick with using a `.extension` scheme (i.e. add a `.`), also others are all lowercase with `-` word seperators